### ISSUE TITLE
Lock icon only shows up in explorer when clicking on the file

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#235208]
Notice that terminalSuggestMain.js does not have the lock icon:

![Image](https://github.com/user-attachments/assets/2565cb77-8131-4c20-b1a2-a2fb383cc56e)

This is because it's showing up only after I open the file. After I close the file it remains, leading to a weird inconsistent state.

**Comment by benibenj** [comment]
@Tyriar you did not use the find widget in this scenario right (just to make sure it was not caused by some recent work)
@lramos15 you aware of this?

**Comment by bpasero** [comment]
I think we could do this but it will also add noise:

![Image](https://github.com/user-attachments/assets/820799fe-1c24-483c-98fa-c1d60ca41d37)

The explorer maintains decorations here:

https://github.com/microsoft/vscode/blob/c5001f6dd570bbdaa082a7f95ade855d5bcbbe17/src/vs/workbench/contrib/files/browser/views/explorerDecorationsProvider.ts#L19

If we decide to really show the 🔒 for all files that are readonly, just add:

```ts
if (fileStat.isReadonly) {
	return {
		letter: Codicon.lockSmall,
		tooltip: localize('readonly', "Read-only"),
	};
}
```

to the end.

**Comment by Tyriar** [comment]
@bpasero how about making it resolved model + open in an editor since the former is invisible to the end user? If you disagree feel free to close.

**Comment by bpasero** [comment]
I think I would either make it consistently showing in all cases or keep what we have now but not add something new that is still not consistent.

**Comment by lramos15** [comment]
Sounds like we can close this as won't fix for now to not do anything confusing since #195701 has been resolved?
